### PR TITLE
Restricting Cross site course enrollment based on SITE organization

### PR DESCRIPTION
**Pull request by ms-manikarthik**
Restricting Cross site course enrollment based on SITE course_org_filter.

### CODE REVIEW

File: common/djangoapps/enrollment/data.py [hunk 1]
```diff
@@ -107,27 +108,30 @@ def create_course_enrollment(username, course_id, mode, is_active):
...
-        raise CourseEnrollmentClosedError(err.message)
-    except CourseFullError as err:
-        raise CourseEnrollmentFullError(err.message)
-    except AlreadyEnrolledError as err:
-        enrollment = get_course_enrollment(username, course_id)
-        raise CourseEnrollmentExistsError(err.message, enrollment)
-
+    current_course_org = course_id.split(':')[1].split('+')[0]
+    site_course_org_filter = configuration_helpers.get_value('course_org_filter')
+    if site_course_org_filter == site_course_org_filter or site_course_org_filter == None:
```
Comment (kabirkhan) [inline common/djangoapps/enrollment/data.py]: Please change `site_course_org_filter == None` to `not site_course_org_filter`

This is generally better practice in python.

**Comment by kabirkhan** [comment]
I understand that but what if we want to use automatic enrollment from a url in the future? Or programmatically enroll users? It's not a huge deal but it is more scalable to allow this addition to be configurable.

**Comment by ms-manikarthik** [comment]
we can still programmatically and automatically enroll user at any point using correct Site name and course id.

**Comment by satyarapelly** [comment]
As per UI/Theme change that we had to restrict the user enroll to course, this fix seems to be okay with me.

**Comment by kabirkhan** [comment]
Approving for now but this is something I think we should revisit. This change will also need corresponding test cases.

File: common/djangoapps/enrollment/data.py [hunk 2]
```diff
@@ -107,27 +108,30 @@ def create_course_enrollment(username, course_id, mode, is_active):
...
-        raise CourseEnrollmentClosedError(err.message)
-    except CourseFullError as err:
-        raise CourseEnrollmentFullError(err.message)
-    except AlreadyEnrolledError as err:
-        enrollment = get_course_enrollment(username, course_id)
-        raise CourseEnrollmentExistsError(err.message, enrollment)
-
+    current_course_org = course_id.split(':')[1].split('+')[0]
+    site_course_org_filter = configuration_helpers.get_value('course_org_filter')
+    if site_course_org_filter == site_course_org_filter or not site_course_org_filter:
```
Comment (sdolenc) [inline common/djangoapps/enrollment/data.py]: nit: just my preference, but i'd flip these conditions. so it'd read
!a || a == b

File: common/djangoapps/enrollment/data.py [hunk 3]
```diff
@@ -107,27 +108,30 @@ def create_course_enrollment(username, course_id, mode, is_active):
...
+            raise CourseNotFoundError(err.message)
+        except EnrollmentClosedError as err:
+            raise CourseEnrollmentClosedError(err.message)
+        except CourseFullError as err:
+            raise CourseEnrollmentFullError(err.message)
+        except AlreadyEnrolledError as err:
+            enrollment = get_course_enrollment(username, course_id)
+            raise CourseEnrollmentExistsError(err.message, enrollment)
+    else:
+        return NonExistentCourseError
```
Comment (sdolenc) [inline common/djangoapps/enrollment/data.py]: unrelated: this is actually how i wanted the original fix to be written. in other words, i'd rather throw this error than conditionalizing some markup in the view. we should consider adopting a similar strategy when we revisit this area soon

File: common/djangoapps/enrollment/data.py (hunk 2 above)
Comment (ms-manikarthik) [inline common/djangoapps/enrollment/data.py]: this would not be the right syntax for python

Comment (sdolenc) [inline common/djangoapps/enrollment/data.py]: obviously. i'm using shorthand

**Pull request merged**